docs/source/services/timelock_service/migration.rst

```diff
@@ -0,0 +1,110 @@
+=======================================
+
+.. danger::
+
+   Improperly configuring one's cluster to use external timestamp and lock services can result in **SEVERE DATA
+   CORRUPTION**! Please contact the AtlasDB team before attempting a migration.
+
+Why Migration?
+--------------
+
+AtlasDB assumes that timestamps returned by the timestamp service are monotonically increasing. In order to preserve
+this guarantee when moving from an embedded timestamp service to an external timestamp service, we need to ensure
+that any timestamps the external timestamp service issues must be larger than any timestamps that the embedded
+service issues. Otherwise, this can lead to serious data corruption.
+
+The migration process must be run offline (that is, with no AtlasDB clients running during migration) and basically
+consists of the following steps:
+
+1. Set up the external timestamp service.
+2. Obtain a timestamp TS that is at least as large as the largest timestamp issued.
+3. Fast-forward the external timestamp service to at least TS.
+4. Invalidate the old timestamp service, preventing AtlasDB clients from retrieving timestamps from it.
+5. Configure AtlasDB clients to retrieve timestamps from the new timestamp service.
+6. Restart AtlasDB clients.
+
+Strictly speaking, step 4 is not required; that said, we prefer to do this to avoid the risk of data corruption in the
+event of misconfiguration (that is, if some but not all clients are pointed to the new timestamp service).
+
+Manual Migration
+----------------
+
+.. danger::
+
+   Ensure that no AtlasDB clients are running during the migration process. Failure to ensure this can result in
+   **SEVERE DATA CORRUPTION** as we are unable to guarantee that timestamps are monotonically increasing!
+
+.. contents::
+   :local:
+
+Step 1: Setting up the Timelock Server
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+First, setup the Timelock Server following the instructions in :ref:`timelock-installation`.
+
+Step 2: Determining the Atlas Timestamp
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+One can use the :ref:`clis` (specifically, the ``timestamp fetch`` command) to obtain a timestamp which is guaranteed
+to be greater than all timestamps issued so far. Make sure that you are requesting for a fresh timestamp as opposed to
+an immutable timestamp (the output should say ``The Fresh timestamp is: N``). Note down the value of this timestamp;
+from here on out, we'll refer to it as N.
+
+Step 3: Fast-Forwarding the Timelock Server
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The Timelock Server exposes an administrative interface, which features a ``fast-forward`` endpoint. Note that this is
+not typically exposed to AtlasDB clients. One can use it to advance the timestamp on the Timelock Server to N, as
+follows:
+
+   .. code:: bash
+
+      curl -XPOST localhost:8080/test/timestamp-admin/fast-forward?newMinimum=N
+
+.. danger::
+
+   Make sure that N has been entered correctly.
```

Comment:
Okay. Basically if N is too large that's okay (apart from `Long.MAX_VALUE` stuff! - though there's a way around that).
If N is too small you need to fast forward again (if clients haven't been started again; if they have, you could do it but you'll already have some data corruption). 

I've mentioned this in the block that follows, but any suggestions on how / if I could make this clearer?